include/msgpack/v1/vrefbuffer.hpp

```diff
@@ -71,6 +71,11 @@ class vrefbuffer {
         m_end   = array + nfirst;
         m_array = array;
 
+        if((sizeof(chunk) + chunk_size) < chunk_size){
+            ::free(array);
+            throw std::bad_alloc();
+        }
+
```

Comment:
Could you move this check to the head of the constructor (line 61) and remove `::free(array);`?

Reply:
OK, done. Also for the C impl, where `vbuf->tail` would point to uinitialized mem in case of overflow.